src/EditorFeatures/TestUtilities/Workspaces/TestHostDocument.cs

```diff
@@ -32,6 +32,7 @@ public class TestHostDocument
         private readonly SourceCodeKind _sourceCodeKind;
         private readonly string _filePath;
         private readonly IReadOnlyList<string> _folders;
+        private readonly IDocumentServiceProvider _documentServiceProvider;
```

Comment:
can this be named 'Opt' since it can be null?  Or can we ensure that if null is passed in, we use a NoOp instance to ensure this is nonnull?